app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,116 @@
+class TasksController < ApplicationController
+  def root
+    @message = 'Hello world!'
+  end
+
+  def show
+    # This is the ruby try-catch.
+    # If Task.find throws an exception
+    # (e.g., can't find a task with that id)
+    # then we can catch the exception and
+    # redirect the user to the homepage with a message.
+    begin
+      @task = Task.find(params['id'])
+    rescue
+      flash[:error] = "Could not find task with id: #{params['id']}" #not printing??? why!
+      redirect_to tasks_path
+      return
+    end
+  end
+
+  def index
+    @tasks = Task.all
+  end
+
+  def create
+    unless params.key?(:task)
+      @task = Task.new
+    else
+      @task = Task.new(
```

Comment:
You have merged the `create` and `new` actions into one controller method. Separating them out like we did in class would make things a little more clear, especially as our controllers get more complex over the coming weeks.